### PR TITLE
Add circleci configuration file 

### LINKED ISSUE #50: Implement a CI to test the build success

### LINKED ISSUE #91: Add Circle CI config File

### PR DESCRIPTION
**Pull request by martinbedouret**
close #91 
close #50

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: cimg/node:18.18.1
+        auth:
+          username: cboardci
+          password: $DOCKERHUB_PASSWORD
+      - image: circleci/mongo:4.0
```
Comment (RodriSanchez1) [inline .circleci/config.yml:10]: Why did you import the `mongo` image?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            - ~/.cache/yarn
+          key: yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      # Build Docker Image
+      # https://circleci.com/docs/2.0/building-docker-images/#overview
+      - setup_remote_docker:   # (2)
+          docker_layer_caching: false  # (3)
+      # build and push Docker image
+      - run: |
+          TAG=0.1.$CIRCLE_BUILD_NUM
+          docker build -t cboard/cboard-ai-builder:$TAG -t cboard/cboard-ai-builder:latest .
```
Comment (RodriSanchez1) [inline .circleci/config.yml:57]: What if  we use:
`docker build -t cboard/cboard-ai-builder:$TAG -t cboard/cboard-ai-builder:latest -f ./docker/Dockerfile .`

In this command, -f ./docker/Dockerfile tells Docker to use the Dockerfile in the ./docker directory, and . sets the build context to the root project directory. This means that COPY and ADD commands in the Dockerfile will be relative to the root project directory, not the ./docker directory.  - From ChatGPT

And with this, we can leave the `DockerFile` inside the docker folder

File: Dockerfile
Comment (RodriSanchez1) [inline Dockerfile:1]: See above comment

**Review by RodriSanchez1 — CHANGES_REQUESTED** [review]
LGTM! 
What do you think that we need to do with the docker compose?. Its located in docker folder.  We can added on this circleci confing?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            # when lock file changes, use increasingly general patterns to restore cache
+            - yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}
+            - yarn-packages-v1-{{ .Branch }}-
+            - yarn-packages-v1-
+      # Install and save node_modules to cache
+      - run: yarn install --cache-folder ~/.cache/yarn
```
Comment (maxicapodacqua) [inline .circleci/config.yml:45]: I'm not sure if this is needed, the docker image is already running `yarn install`

File: Dockerfile
Comment (maxicapodacqua) [inline Dockerfile:1]: This will break the current docker compose file, consider updating it or update the circleci to use the current directory

**Review by maxicapodacqua — CHANGES_REQUESTED** [review]

**Pull request merged**